scripts/Inelastic/CrystalField/fitting.py

```diff
@@ -1444,6 +1553,59 @@ def _fit_multi(self):
         self.model.update(function)
         self.model.chi2 = alg.getProperty('OutputChi2overDoF').value
 
+    def fit_sp(self, Solver: str, Options: dict = None) -> None:
+        """
+        Run scipy.optimize.minimize algorithm for CEF parameters only. Update function parameters.
+        """
+        self.check_consistency()
+        from CrystalField.CrystalFieldMultiSite import CrystalFieldMultiSite
+        if isinstance(self.model, CrystalFieldMultiSite):
+            fun = self.model.function
+        else:
+            fun = self._function
+        cef_fixed = []
+        for par_id in [id for id in range(fun.nParams()) if fun.isFixed(id)]:
+            parName = fun.getParamName(par_id)
+            if parName in CrystalField.field_parameter_names or "IntensityScaling" in parName:
+                cef_fixed.append(par_id)
+        x0 = []
+        fun.setAttributeValue('FixAllPeaks', True)
+        for par_id in self._free_cef_parameters:
+            x0.append(fun.getParameterValue(par_id))
+            fun.fixParameter(par_id)
+        opt = {'disp': False}
+        if Options is not None:
+            opt = Options
+        for (x, pos) in zip(x0, self._free_cef_parameters):
+            fun.removeTie(pos)
+            res = sp.minimize(self._evaluate_cf, [x], args=(fun, pos), method=Solver, options=opt)
+            fun.fixParameter(pos)
+            if res.success:
+                if self._function is not None:
+                    if isinstance(res.x, list):
+                        fun.setParameter(pos, float(res.x[0]))
+                    else:
+                        fun.setParameter(pos, float(res.x))
+        self.model.update(fun)
+
+    def _evaluate_cf(self, x0: float, fun: IFunction, cef_pos: int) -> float:
+        from mantid.simpleapi import CalculateChiSquared
+        fun.setParameter(cef_pos, x0[0])
+        if isinstance(self._input_workspace, list):
+            ws_kwargs = {}
+            ws_kwargs['InputWorkspace'] = self._input_workspace[0]
+            i = 1
+            for workspace in self._input_workspace[1:]:
+                ws_kwargs['InputWorkspace_{}'.format(i)] = workspace
+                i += 1
+            fun_str = re.sub(r'FWHM[X|Y]\d+=\(\),', '', str(fun))
+            fun_str = re.sub(r'(name=.*?,)(.*?)(Temperatures=\(.*?\),)',r'\1\3\2', fun_str)
+            fun_str = re.sub(r'(name=.*?,)(.*?)(PhysicalProperties=\(.*?\),)',r'\1\3\2', fun_str)
+            fun_str = re.sub(r'f[0-9]+\.f(['+str(fun.getAttributeValue("MaxPeakCount"))+r'-9]|[1-9][0-9])\.\w+=.*?,','', fun_str)
```

Comment:
Regexes are "fun" to read after a while, could we comment one example of a sample input for future developers:

E.g.
```python
# FWHMY=1  (N.b. I'm just guessing mid-review what this would be)
fun_str = re.sub(r'FWHM[X|Y]\d+=\(\),', '', str(fun))
```